src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+
+        ColorSupplier supplier = new ColorSupplier();
+        Random random = new Random();
+        int boundNumber = 15;
+
+        Figure[] figures = {
+                new Circle(
+                        supplier.getRandomColor(),
+                        "Circle",
+                        random.nextInt(boundNumber)
+                ),
+                new Square(
+                        supplier.getRandomColor(),
+                        "Square",
+                        random.nextInt(boundNumber)
+                ),
+                new Rectangle(
+                        supplier.getRandomColor(),
+                        "Rectangle", random.nextInt(boundNumber),
+                        random.nextInt(boundNumber)
+                ),
+                new RightTriangle(
+                        supplier.getRandomColor(),
+                        "Triangle", random.nextInt(boundNumber),
+                        random.nextInt(boundNumber)
+                ),
+                new IsoscelesTrapezoid(
+                        supplier.getRandomColor(),
+                        "Trapezoid",
+                        random.nextInt(boundNumber),
+                        random.nextInt(boundNumber),
+                        random.nextInt(boundNumber)
+                )
+        };
+        return figures[new Random().nextInt(figures.length)];
```

Comment:
lets make a switch-case
you are creating a lot of redundant figures each time this method is invoked - you only need 1 - but create 5

```
int number = random.nextInt(FIGURE_NUMBER);
switch (number) {
    case 0:
        return getRandomSquare();
    case 1:
        ....
}
```